src/sympc/protocol/falcon/falcon.py

```diff
@@ -63,3 +65,77 @@ def __eq__(self, other: Any) -> bool:
             return False
 
         return True
+
+    @staticmethod
+    def mul(x, y, session) -> ReplicatedSharedTensor:
+        """Master method for multiplication.
+
+        Performs Falcon's mult implementation, gets and reshares mul results and distributes shares.
+        This needs to be improved in future, it relies on orchestrator being a trusted third party.
+        Falcon, requires parties to be able to communication between each other.
+
+        Args:
+            x (MPCTensor): Secret
+            y (MPCTensor): Another secret
+            session (Session): Session the tensors belong to
+
+        Returns:
+            shares (ReplicatedSharedTensor): Shares in terms of ReplicatedSharedTensor.
+
+        Raises:
+            ValueError: Raised when number of parties are not three.
+            NotImplementedError: Raised when implementation not present
+
+        """
+        if not len(session.parties) == 3:
+            raise ValueError("Falcon requires 3 parties")
+
+        result = None
+
+        if session.protocol.security_type == "semi-honest":
+            result = Falcon.mul_implementation(x, y)
+
+        elif session.protocol.security_type == "malicious":
+            raise NotImplementedError(
+                "Multiplication has not been implemented for malicious setting"
+            )
+
+        else:
+            raise NotImplementedError(
+                "Multiplication has not been implemented for "
+                + session.protocol.security_type
+                + " setting"
+            )
+
+        vals = [result[0].get(), result[1].get(), result[2].get()]
+        # Add random mask to vals
+        shares = MPCTensor.generate_shares(
```

Comment:
Q: Why are we not using PRZS to 3-out-3 sharing?

Reply:
Dint see the difference. PRZS is effective only for party-party communication. Here PRZS will be essentially the same as generating shares.

Reply:
True , since we rely on the orchestrator model,it does not make a difference.From a security point, falcon does not depend on third party.If we could generate local from the parties, it would be consistent with Falcon Hrishi.

Reply:
Yes, but if the orchestrator has access to all the shares. The orchestrator can anyway sum it up?

Reply:
True , I totally agree,but that assumption is due to the current sympc orchestrator model. It changes  the building blocks of Falcon from security standpoint Hrishi,so I think generating from PRZS would be a better option.

Reply:
Okay, I will be implementing it since we don't have to change code when we move P2P someday.